### DISCUSSION TITLE
Is it possible to have recursive routes

### DISCUSSION THREAD

**Discussion by ecyrbe** [TanStack/router#302] [Q&A]
Is it possible for react location to handle these Dynamic routes: 

- `/page/:id1/..../idN`
- `/page/:id1/..../idN/subpage`
- `/page/:id1/..../idN/subpage/:pageid`
....
Where `N` is known only at runtime?

**Comment by ondrejvelisek** [comment]
Would love to have this too. Is it possible? Cant find anything on that in docs.

**Comment by schiller-manuel** [comment]
you can implement this yourself via splat routes 

https://tanstack.com/router/v1/docs/framework/react/guide/routing-concepts#splat--catch-all-routes

**Reply by mx-bernhard** [reply]
This is now at this location, right?

https://tanstack.com/router/v1/docs/framework/react/routing/code-based-routing#splat--catch-all-routes

**Reply by schiller-manuel** [reply]
https://tanstack.com/router/latest/docs/framework/react/routing/routing-concepts#splat--catch-all-routes
